src/cmake/cmake-executable.ts

```diff
@@ -9,13 +9,13 @@ export interface CMakeExecutable {
   minimalServerModeVersion: util.Version;
 }
 
-export async function getCMakeExecutableInformation(path: string): Promise<CMakeExecutable> {
+export async function getCMakeExecutableInformation(path: string|null): Promise<CMakeExecutable> {
```

Comment:
Keep this accepting a `string` and instead `console.assert(path.length != 0)`, that is to prevent a bad executable from going through. Change code that gets the path from the config to handle the error at that level, rather than at this level.

Reply:
I would change it to accept only string.
The assert has the problem it will rise an `AssertException`, this would broke the control flow, but this function should provide a valid executable information by returning the cmake struct. In cause of a problem it would be `cmake.isPresent = false`, and the executable is invalid.